pkg/agent/datamodel/types.go

```diff
@@ -616,55 +616,30 @@ type SysctlConfig struct {
-	Role                                AgentPoolProfileRole `json:"role,omitempty"`
-	AcceleratedNetworkingEnabled        *bool                `json:"acceleratedNetworkingEnabled,omitempty"`
-	AcceleratedNetworkingEnabledWindows *bool                `json:"acceleratedNetworkingEnabledWindows,omitempty"`
-	VMSSOverProvisioningEnabled         *bool                `json:"vmssOverProvisioningEnabled,omitempty"`
-	FQDN                                string               `json:"fqdn,omitempty"`
-	CustomNodeLabels                    map[string]string    `json:"customNodeLabels,omitempty"`
-	PreprovisionExtension               *Extension           `json:"preProvisionExtension"`
-	Extensions                          []Extension          `json:"extensions"`
-	KubernetesConfig                    *KubernetesConfig    `json:"kubernetesConfig,omitempty"`
-	OrchestratorVersion                 string               `json:"orchestratorVersion"`
-	ImageRef                            *ImageReference      `json:"imageReference,omitempty"`
-	MaxCount                            *int                 `json:"maxCount,omitempty"`
-	MinCount                            *int                 `json:"minCount,omitempty"`
-	EnableAutoScaling                   *bool                `json:"enableAutoScaling,omitempty"`
-	AvailabilityZones                   []string             `json:"availabilityZones,omitempty"`
-	PlatformFaultDomainCount            *int                 `json:"platformFaultDomainCount"`
-	PlatformUpdateDomainCount           *int                 `json:"platformUpdateDomainCount"`
-	SinglePlacementGroup                *bool                `json:"singlePlacementGroup,omitempty"`
-	VnetCidrs                           []string             `json:"vnetCidrs,omitempty"`
-	PreserveNodesProperties             *bool                `json:"preserveNodesProperties,omitempty"`
-	WindowsNameVersion                  string               `json:"windowsNameVersion,omitempty"`
-	EnableVMSSNodePublicIP              *bool                `json:"enableVMSSNodePublicIP,omitempty"`
-	LoadBalancerBackendAddressPoolIDs   []string             `json:"loadBalancerBackendAddressPoolIDs,omitempty"`
-	CustomVMTags                        map[string]string    `json:"customVMTags,omitempty"`
-	DiskEncryptionSetID                 string               `json:"diskEncryptionSetID,omitempty"`
-	UltraSSDEnabled                     *bool                `json:"ultraSSDEnabled,omitempty"`
-	EncryptionAtHost                    *bool                `json:"encryptionAtHost,omitempty"`
-	ProximityPlacementGroupID           string               `json:"proximityPlacementGroupID,omitempty"`
-	CustomKubeletConfig                 *CustomKubeletConfig `json:"customKubeletConfig,omitempty"`
-	CustomLinuxOSConfig                 *CustomLinuxOSConfig `json:"customLinuxOSConfig,omitempty"`
+	Name                   string               `json:"name"`
+	Count                  int                  `json:"count"`
```

Comment:
question: for this PR, we are only removing parameters that are not populated in `paramsMap` - correct?

I could be wrong but I think `Count` is one of those that get written to the map but aren't actually used anywhere.

Reply:
Thanks for the info. Yeah, you are right, this PR only removes the easy, obvious ones. Once this goes in, I'll look into "Count". If as you mentioned it is not used, I'll remove it.